activation/poet.go

```diff
@@ -45,8 +45,26 @@ type PoetAuth struct {
 	*certifier.PoetCert
 }
 
+type PoetBackend interface {
```

Comment:
I'm not sure I like the naming. This should be called `PoetClient`, while the current `PoetClient` interface should be called something like `PoetService` or similar, since this is basically just the http client (or a mock of it) while the other interface/implementation has additional logic inside.

Wdyt?